eth/block-validation/api.go

```diff
@@ -344,3 +346,89 @@ func (b *BlockAssemblerRequest) UnmarshalJSON(data []byte) error {
 
 	return nil
 }
+
+func (api *BlockValidationAPI) BlockAssembler(params *BlockAssemblerRequest) (*capella.ExecutionPayload, error) {
+	log.Info("BlockAssembler", "tobTxs", len(params.TobTxs.Transactions), "robPayload", params.RobPayload)
+	transactionBytes := make([][]byte, len(params.TobTxs.Transactions))
+	for i, txHexBytes := range params.TobTxs.Transactions {
+		transactionBytes[i] = txHexBytes[:]
+	}
+	decodedTobTxs, err := engine.DecodeTransactions(transactionBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	robBlock, err := engine.ExecutionPayloadV2ToBlock(params.RobPayload.ExecutionPayload)
+	if err != nil {
+		return nil, err
+	}
+
+	tobTxs := types.Transactions(decodedTobTxs)
+
+	// TODO - check for gas limits
+	// TODO - support for payouts
+
+	// TODO - if there are no TOB txs then we can just simulate the block rather then re-assembling it.
+	// These are TODOs for now to simplify integration testing
+
+	// check if there are any duplicate txs
+	// we can error out if there is a nonce gap
+	// TODO - don't error out, but drop the duplicate tx in the ROB block
+	seenTxMap := make(map[common.Hash]struct{})
```

Comment:
why is the type of the map an empty struct? bool seems clearer

Reply:
true it is cleaner. I have seen this in the current code with mempoolHashes but bool just makes more sense yes